### DISCUSSION TITLE
Multithreading  Feature

### DISCUSSION THREAD

**Discussion by yongchin0821** [SeldomQA/seldom#171] [General]
## seldom 多线程问题讨论

在群里已经多次看到群友对于多线程的需求。

首先说一下个人的观点，多线程其实是一个公共需求（比如用多线程进行报文发送，并发抽取数据，并发调度启动任务，并发http请求进行性能测试等）。针对这种需求，python本来就自带了多线程的能力，换言之就是自带多线程的库。而这种需求往往很少进行定制化为**某个模块或者框架**自己独有的能力。

若只是**纯粹地把多线程和seldom缝合在一起。那是否有必要就值得斟酌。** 除非真的对seldom有一定的理解，做好封装的同时提供出的API 优于python自带多线程那才是有意义的。

对于是否需要多线程我觉得这还需要考虑或者说更多的是取决于**功能边界**。seldom主要是用于自动化测试这一块。多线程其实是可有可无，除非有特别的定制化需求，而seldom作为一个自动化测试框架，除了提供便利的公共能力外，也没法很好的cover住各种特定多线程场景。特殊多线程场景还得是使用者自己来适配。这样才比较灵活。毕竟越定制化就越不自由。我们始终要在高定制化和高自由度中找到平衡。同时，从广义上来讲，多线程本来就可以启动seldom，不仅可以启用1个。还可以启动2个，3个。。。

**Comment by defnngj** [comment]
你想复杂了，我想大家想要的 多线程 是 用例可以并行。

当然，并发执行可以和 seldom 没关系。例如： https://github.com/SeldomQA/seldom/issues/143

但是，上面的例子相当于 `不同的线程 跑不同的用例 生成多份报告`，或者 `不同的线程 跑多遍一样的用例 生成多份报告`。

大家真正想要的是： `设置多线程 跑不同的用例，最后生成一份报告`。 这样可以有效缩短用例执行时间。 和具体什么业务无关。
